src/typing.py

```diff
@@ -1896,7 +1896,8 @@ def _make_nmtuple(name, types):
     msg = "NamedTuple('Name', [(f0, t0), (f1, t1), ...]); each t must be a type"
     types = [(n, _type_check(t, msg)) for n, t in types]
     nm_tpl = collections.namedtuple(name, [n for n, t in types])
-    nm_tpl._field_types = dict(types)
+    nm_tpl.__annotations__ = collections.OrderedDict(types)
+    nm_tpl._field_types = nm_tpl.__annotations__
```

Comment:
I would prefer to write this as a single assignment ``nm_tpl.__annotations__ = nm_tpl._field_types = collections.OrderedDict(types)`` and add a comment explaining that we have two names for pre-/post-PEP526 compatibility.